main.go

```diff
@@ -221,8 +226,8 @@ func Main() int {
 			srv, err = LocalForwardUDP(&state, s, lf, doneChannel)
 		}
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "[!] Failed to listen on %s://%s:%d: %s\n",
-				lf.network, lf.bind.Addr, lf.bind.Port, err)
+			failNow(fmt.Sprintf("[!] Failed to listen on %s://%s:%d: %s\n",
+				lf.network, lf.bind.Addr, lf.bind.Port, err))
```

Comment:
That begs for a test. Bind to a socket, run slirp hitting the same port. See the unique exit code.